101-iothub-auto-route-messages/azuredeploy.json

```diff
@@ -0,0 +1,172 @@
+            "defaultValue": "demostorage",
+            "maxLength": 11,
+            "metadata": {
+                "description": "Name prefix of storage account to be created."
+            }
+        },
+        "storageContainerName": {
+            "type": "string",
+            "defaultValue": "contosoresults",
+            "metadata": {
+                "description": "Name of the container in which to place the routed data."
+            }
+        },
+        "storageEndpoint": {
+            "type": "string",
+            "defaultValue": "ContosoStorageEndpoint",
+            "metadata": {
+                "description": "Name of the endpoint for the storage account."
+            }
+        }
+    },
+    "variables": {
+        "iotHubName": "[concat(parameters('IotHubNamePrefix'), uniqueString(resourceGroup().id))]",
+        "storageAccountName": "[concat(parameters('storageAccountNamePrefix'), uniqueString(resourceGroup().id))]"
+    },
+    "resources": [
+        {
+            "type": "Microsoft.Storage/storageAccounts",
+            "apiVersion": "2019-06-01",
+            "name": "[variables('storageAccountName')]",
+            "location": "[parameters('location')]",
+            "sku": {
+                "name": "Standard_LRS",
+                "tier": "Standard"
+            },
+            "kind": "Storage",
+            "properties": {},
+            "resources": [
+                {
+                    "type": "blobServices/containers",
+                    "apiVersion": "2019-06-01",
+                    "name": "[concat('default/', parameters('storageContainerName'))]",
+                    "properties": {
+                        "publicAccess": "None"
+                    },
+                    "dependsOn": [
+                        "[resourceId('Microsoft.Storage/storageAccounts', variables('storageAccountName'))]"
+                    ]
+                }
+            ]
+        },
+        {
+            "type": "Microsoft.Devices/IotHubs",
+            "apiVersion": "2020-03-01",
+            "name": "[variables('IoTHubName')]",
+            "location": "[parameters('location')]",
+            "dependsOn": [
+                "[resourceId('Microsoft.Storage/storageAccounts', variables('storageAccountName'))]"
+            ],
+            "properties": {
+                "eventHubEndpoints": {
+                    "events": {
+                        "retentionTimeInDays": 1,
+                        "partitionCount": "[parameters('d2cPartitions')]"
+                    }
+                },
+                "routing": {
+                    "endpoints": {
+                        "storageContainers": [
+                            {
+                                "connectionString": "[Concat('DefaultEndpointsProtocol=https;AccountName=',variables('storageAccountName'),';AccountKey=',listKeys(resourceId('Microsoft.Storage/storageAccounts', variables('storageAccountName')), '2019-06-01').keys[0].value)]",
```

Comment:
this needs to account for fairfax, it seems the RP is hardcoding the endpoint suffix to public cloud... you can use the environment() function to get the cloud specific suffix